mlrun/datastore/azure_blob.py

```diff
@@ -135,3 +136,69 @@ def listdir(self, key):
     def rm(self, path, recursive=False, maxdepth=None):
         path = self._convert_key_to_remote_path(key=path)
         super().rm(path=path, recursive=recursive, maxdepth=maxdepth)
+
+    def get_spark_options(self):
+        res = {}
+        st = self.get_storage_options()
+        service = "blob"
+        primary_url = None
+        if st.get("connection_string"):
+            primary_url, _, parsed_credential = parse_connection_str(
+                st.get("connection_string"), credential=None, service=service
+            )
+            for key in ["account_name", "account_key"]:
+                if parsed_credential.get(key):
+                    if st[key] and st[key] != parsed_credential.get(key):
+                        if key == "account_name":
+                            raise mlrun.errors.MLRunInvalidArgumentError(
+                                f"'{key}' from the storage_options does not match corresponding connection string - \
+                                    {st[key]} != {parsed_credential.get(key)}"
```

Comment:
```suggestion
                                f"Storage option for '{key}' is '{st[key]}', which does not match corresponding connection string '{parsed_credential.get(key)}'"
```
Maybe this is a bit clearer.